app/services/webhook/props.go

```diff
@@ -0,0 +1,161 @@
+	case enum.WebhookDeletePost:
+		DescribePost(props, dummyPost, "post", baseURL, true, true)
+		props["post_status"] = enum.PostDeleted
+		props["post_response_text"] = "The reason _why_ this post was deleted."
+	}
+	return props
+}
+
+func DescribeUser(props dto.Props, user *entity.User, keyPrefix string) {
+	if user == nil {
+		return
+	}
+	role, _ := user.Role.MarshalText()
+	props.Append(dto.Props{
+		keyPrefix + "_id":     user.ID,
+		keyPrefix + "_name":   user.Name,
+		keyPrefix + "_email":  user.Email,
+		keyPrefix + "_role":   string(role),
+		keyPrefix + "_avatar": user.AvatarURL,
+	})
+}
+
+func DescribePost(props dto.Props, post *entity.Post, keyPrefix, baseURL string, complete, author bool) {
+	if post == nil {
+		return
+	}
+	props.Append(dto.Props{
+		keyPrefix + "_id":          post.ID,
+		keyPrefix + "_number":      post.Number,
+		keyPrefix + "_title":       post.Title,
+		keyPrefix + "_slug":        post.Slug,
+		keyPrefix + "_description": post.Description,
+		keyPrefix + "_created_at":  post.CreatedAt,
+		keyPrefix + "_url":         fmt.Sprintf("%s/posts/%d/%s", baseURL, post.Number, post.Slug),
+	})
+	if author {
+		DescribeUser(props, post.User, keyPrefix+"_author")
+	}
+	if complete {
+		postResponse := post.Response
+		props.Append(dto.Props{
+			keyPrefix + "_votes":    post.VotesCount,
+			keyPrefix + "_comments": post.CommentsCount,
+			keyPrefix + "_status":   post.Status.Name(),
+			keyPrefix + "_tags":     post.Tags,
+			keyPrefix + "_response": postResponse != nil,
+		})
+		if postResponse != nil {
+			keyPrefix := keyPrefix + "_response"
+			props.Append(dto.Props{
+				keyPrefix + "_text":         postResponse.Text,
+				keyPrefix + "_responded_at": postResponse.RespondedAt,
+			})
+			DescribeUser(props, postResponse.User, keyPrefix+"_author")
+			originalPost := postResponse.Original
+			if post.Status == enum.PostDuplicate && originalPost != nil {
+				keyPrefix := keyPrefix + "_original"
+				props.Append(dto.Props{
+					keyPrefix + "_number": originalPost.Number,
+					keyPrefix + "_title":  originalPost.Title,
+					keyPrefix + "_slug":   originalPost.Slug,
+					keyPrefix + "_status": originalPost.Status.Name(),
+					keyPrefix + "_url":    fmt.Sprintf("%s/posts/%d/%s", baseURL, originalPost.Number, originalPost.Slug),
+				})
+			}
+		}
+	}
+}
+
+func DescribeTenant(props dto.Props, tenant *entity.Tenant, keyPrefix, baseURL, logoURL string) {
+	if tenant == nil {
+		return
+	}
+	props.Append(dto.Props{
+		keyPrefix + "_id":              tenant.ID,
+		keyPrefix + "_name":            tenant.Name,
+		keyPrefix + "_subdomain":       tenant.Subdomain,
+		keyPrefix + "_invitation":      tenant.Invitation,
+		keyPrefix + "_welcome_message": tenant.WelcomeMessage,
+		keyPrefix + "_status":          TenantStatus(tenant.Status),
+		keyPrefix + "_locale":          tenant.Locale,
+		keyPrefix + "_url":             baseURL,
+		keyPrefix + "_logo":            logoURL,
+	})
+}
+
+func TenantStatus(status int) string {
```

Comment:
would be better to implement MarshalText/UnmarshalText/String on the enum and use `tenant.Status.String()`

all enums should have this now, but it's possible that I missed some as I added this late

Reply:
I didn't find any method to get a string representation, that's why I did mine, but it's true it would be better to add one instead!